proxy-node-shared/src/test/java/uk/gov/ida/notification/shared/proxy/VSPProxyTranslateResponseTest.java

```diff
@@ -0,0 +1,149 @@
+package uk.gov.ida.notification.shared.proxy;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.dropwizard.testing.junit.DropwizardClientRule;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.ida.exceptions.ApplicationException;
+import uk.gov.ida.jerseyclient.ErrorHandlingClient;
+import uk.gov.ida.jerseyclient.JsonClient;
+import uk.gov.ida.jerseyclient.JsonResponseProcessor;
+import uk.gov.ida.notification.contracts.verifyserviceprovider.AuthnRequestGenerationBody;
+import uk.gov.ida.notification.contracts.verifyserviceprovider.TranslatedHubResponse;
+import uk.gov.ida.notification.contracts.verifyserviceprovider.TranslatedHubResponseBuilder;
+import uk.gov.ida.notification.contracts.verifyserviceprovider.VerifyServiceProviderTranslationRequest;
+import uk.gov.ida.notification.contracts.verifyserviceprovider.VspLevelOfAssurance;
+import uk.gov.ida.notification.contracts.verifyserviceprovider.VspScenario;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.eq;
+
+@RunWith(MockitoJUnitRunner.class)
+public class VSPProxyTranslateResponseTest {
+
+    @Path("/")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public static class TestVSPResource {
+
+        @POST
+        @Path("/translate-response")
+        public TranslatedHubResponse testGetTranslatedHubResponse(VerifyServiceProviderTranslationRequest request) {
+            if ("SAMLResponse1234".equals(request.getSamlResponse())) {
```

Comment:
Can we have two separate resources to make the authn successful vs failed scenarios explicit rather than switching on an arbitrary string value?

Reply:
This keeps the tests for each path on VspProxy separate.  It looked a bit messy when they were both in together.

Reply:
Done.